### PR TITLE
Adds If Not Exists to Create Table As

### LINKED ISSUE #14834: Add IF NOT EXISTS support to CREATE TABLE AS
### CrateDB version

nightly

### CrateDB setup information

```
docker run --rm -it --name=cratedb --publish=4200:4200 --publish=5432:5432     --env=CRATE_HEAP_SIZE=4g crate/crate:nightly
```

### Problem description

When using the `CREATE TABLE ... AS ...` statement, you can not use the `IF NOT EXISTS` clause.

### Steps to Reproduce

```sql
CREATE TABLE IF NOT EXISTS foobar AS SELECT 1;
```

### Actual Result

```
SQLParseException[line 1:36: mismatched input 'AS' expecting {'.', '('}]
```

### Expected Result

```
CREATE OK, 1 row affected  (0.091 sec)
```

### References

- https://github.com/crate-workbench/cratedb-toolkit/pull/56#pullrequestreview-1667420193

### PR DESCRIPTION
**Pull request by dshunter107**
Closes #14834

## Checklist

 - [ ] Added an entry in the latest `docs/appendices/release-notes/<x.y.0>.rst` for user facing changes
 - [ ] Updated documentation & `sql_features` table for user facing changes
 - [] Touched code is covered by tests
 - [x ] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [ ] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in the latest `docs/appendices/release-notes/<x.y.0>.rst`
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by dshunter107** [comment]
Hello. I am adding to this a little at a time. I will continue with adding to the Analyzer and the Integration Tests.

In the meantime, I have questions about tests. I could not find any tests for Create Table As in TestStatementBuilder.java, so I made a function. Are there already tests for Create Table As for the parsing portion of the code. And if so, where are they? Also, should I move my tests to another location?

**Comment by matriv** [comment]
ok to test

File: libs/sql-parser/src/main/java/io/crate/sql/SqlFormatter.java [hunk 1]
```diff
@@ -623,6 +624,22 @@ public Void visitCreateTable(CreateTable node, Integer indent) {
             return null;
         }
 
+        @SuppressWarnings({ "rawtypes", "unchecked" })
```
Comment (matriv) [inline libs/sql-parser/src/main/java/io/crate/sql/SqlFormatter.java:627]: Remove this please, and see below.

File: libs/sql-parser/src/main/java/io/crate/sql/SqlFormatter.java [hunk 2]
```diff
@@ -623,6 +624,22 @@ public Void visitCreateTable(CreateTable node, Integer indent) {
...
+        @Override
+        public Void visitCreateTableAs(CreateTableAs node, Integer indent) {
```
Comment (matriv) [inline libs/sql-parser/src/main/java/io/crate/sql/SqlFormatter.java:629]:
```suggestion
        public Void visitCreateTableAs(CreateTableAs<?> node, Integer indent) {
```

File: libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java [hunk 3]
```diff
@@ -852,6 +852,12 @@ public void testCreateTableOptionsMultipleTimesNotAllowed() {
             .hasMessage("line 1:83: mismatched input 'partitioned' expecting {<EOF>, ';'}");
     }
 
+    @Test
+    public void testCreateTableAs() {
```
Comment (matriv) [inline libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java:856]: Please move this below: `testCreateTableStmtBuilder()`, and move the `create table ... as` lines in `testCreateTableStmtBuilder` into this method, thx!

**Review by matriv — COMMENTED** [review]
Thx for starting working on this @dshunter107!

Left some comments already for the parser, and in order to make use and actually test the SqlFormatter please add under this line: https://github.com/crate/crate/blob/master/libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java#L2132
```
statement instanceof CreateTableAs ||
```

Thx!

**Comment by dshunter107** [comment]
Hello, 
Just giving a quick update. I am applying to some mentorships for spring, and they involve doing some tasks. As soon as I finish the last one, I will get back to this. The last one is due in a few days.

**Comment by matriv** [comment]
> Hello, Just giving a quick update. I am applying to some mentorships for spring, and they involve doing some tasks. As soon as I finish the last one, I will get back to this. The last one is due in a few days.

Thx for updating us on this @dshunter107!

**Comment by matriv** [comment]
any progress here @dshunter107 ?

File: libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java (hunk 3 above)
Comment (dshunter107) [reply libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java:862]: done. To clarify. You are saying that I should get rid of the testCreateTableAs function, and put the lines in the testCreateTableStmtBuilder, correct?

File: libs/sql-parser/src/main/java/io/crate/sql/SqlFormatter.java (hunk 2 above)
Comment (dshunter107) [reply libs/sql-parser/src/main/java/io/crate/sql/SqlFormatter.java:683]: done

File: libs/sql-parser/src/main/java/io/crate/sql/SqlFormatter.java (hunk 1 above)
Comment (dshunter107) [reply libs/sql-parser/src/main/java/io/crate/sql/SqlFormatter.java:681]: done

File: libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java (hunk 3 above)
Comment (matriv) [reply libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java:862]: nope, just move it below the `testCreateTableStmtBuilder()`, to be close together,
and the tests inside `testCreateTableStmtBuilder()` which test the `create table AS` move them to your new method: `testCreateTableAs`, thx!

Comment (matriv) [reply libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java:862]: Apologies, there are no `CREATE TABLE<tableName> AS` stmts in this method, please ignore that, just add your new method below that one, all fine, and again apologies, I didn't check carefully.

Comment (dshunter107) [reply libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java:862]: Okay sounds good. No worries

**Comment by dshunter107** [comment]
okay. this was a disaster. Sorry for this. I was trying to figure out why my branch had changes to over 400 files, so i performed checkout on the most recent commit by someone else. I then pushed HEAD to see if I could get back to only showing the 5 files that I changed, and instead, it got rid of everything and is saying that it successfully merged and closed. I will attempt again.

**Comment by matriv** [comment]
Superseded by: https://github.com/crate/crate/pull/15611

**Pull request closed without merging**